lib/person.rb

```diff
@@ -0,0 +1,12 @@
+class Person
+  attr_reader :name, :interests, :supplies
+  def initialize(attributes)
+    @name = attributes[:name]
+    @interests = attributes[:interests]
+    @supplies = {}
+  end
+
+  def add_supply(supply, amount)
+    @supplies[supply] = amount
```

Comment:
You are really close here. Unfortunately the way you have this currently will override the value of an existing supply instead of increase the amount of that supply. Consider doing something like this instead?

```
if @supplies[supply]
  @supplies[supply] += amount
else
  @supply[supply] = amount
end
```

```